admin/cmd/bootstrap.go

```diff
@@ -211,3 +218,46 @@ func processFinalError(err error, isJSON bool) {
 		os.Exit(1)
 	}
 }
+
+func checkVersionCompatibility(jsonOutput bool) {
+	clientVersion := version.Version
+	serverStatus, err := agentlocal.GetStatus(agentlocal.DoNotRequestNetworkInfo)
+	if err != nil {
+		return
+	}
+
+	if err = compareVersions(clientVersion, serverStatus.ServerVersion); err != nil {
+		if jsonOutput {
+			b, jErr := json.Marshal(err.Error())
+			if jErr != nil {
+				logrus.Infof("Error: %#v.", err)
+				logrus.Panicf("Failed to marshal error to JSON.\n%s.\nPlease report this bug.", jErr)
+			}
+			fmt.Printf("%s\n", b) //nolint:forbidigo
+		} else if errors.Is(err, errMismatchedVersion) {
+			fmt.Println(err) //nolint:forbidigo
+		} else {
+			logrus.Infof("Failed to check version compatibility: %s.", err.Error())
+		}
```

Comment:
Can we use `logrus` instead of `fmt` here?
It appears it shall be fully initialized by this point, including JSON formatting.